debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -942,30 +960,13 @@ public void readSchema(Tables tables, String databaseCatalog, String schemaNameP
                     continue;
                 }
                 if (tableFilter == null || tableFilter.isIncluded(tableId)) {
-                    List<Column> cols = columnsByTable.computeIfAbsent(tableId, name -> new ArrayList<>());
-                    String columnName = rs.getString(4);
-                    if (columnFilter == null || columnFilter.matches(catalogName, schemaName, tableName, columnName)) {
-                        ColumnEditor column = Column.editor().name(columnName);
-                        column.jdbcType(rs.getInt(5));
-                        column.type(rs.getString(6));
-                        column.length(rs.getInt(7));
-                        if (rs.getObject(9) != null) {
-                            column.scale(rs.getInt(9));
-                        }
-                        column.optional(isNullable(rs.getInt(11)));
-                        column.position(rs.getInt(17));
-                        column.autoIncremented("YES".equalsIgnoreCase(rs.getString(23)));
-                        String autogenerated = null;
-                        try {
-                            autogenerated = rs.getString(24);
-                        } catch (SQLException e) {
-                            // ignore, some drivers don't have this index - e.g. Postgres
-                        }
-                        column.generated("YES".equalsIgnoreCase(autogenerated));
-
-                        column.nativeType(resolveNativeType(column.typeName()));
-
-                        cols.add(column.create());
+                    final List<Column> cols = readTableColumns(rs, tableId, columnFilter).stream().map(ColumnEditor::create).collect(Collectors.toList());
```

Comment:
Can you put each stream operation to a new line? Otherwise it's a nightmare to debug.